### ISSUE TITLE
Accessing non-existing self.action

### ISSUE THREAD

**Issue by remod** [leggedrobotics/free_gait#60]
When starting move_base.py in autonomous mode, I sometimes get the following error:

```
[ INFO] [1528186876.028398748, 138.164000000]: [mission_states::RosInterface::publishProtocolEntry] PoseAdaptation: Started. ()
[ INFO] [1528186876.031453179, 138.164000000]: Worker [switch_from_freeze_to_free_gait_impedance_ros] started
[ WARN] [1528186876.031827301, 138.164000000]: [tinyxml_tools]: Could not find attribute 'maximalNormalForce' in element: <Constraints minimalNormalForce="15.0" />. Setting to default value '1.79769e+308'.
[ WARN] [1528186876.031999135, 138.164000000]: [tinyxml_tools]: Could not find attribute 'maximalNormalForce' in element: <Constraints minimalNormalForce="15.0" />. Setting to default value '1.79769e+308'.
[ WARN] [1528186876.033489078, 138.165000000]: NEW initialize
[ WARN] [1528186876.106990440, 138.170000000]: Key '/loco/ContactInvariantDamper/headingDampingGain' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.107170117, 138.170000000]: Key '/loco/ContactInvariantDamper/lateralDampingGain' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.107457787, 138.170000000]: Key 'vmcKpLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.107617453, 138.170000000]: Computation of high-level controller is not real-time! Computation took: 31.304256 ms. (Warning is throttled: 3s)
[ WARN] [1528186876.108376543, 138.170000000]: Key 'vmcKdLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.108527933, 138.170000000]: Key 'vmcKiLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.108684622, 138.170000000]: Key 'vmcKffLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.108843062, 138.170000000]: Key 'vmcKpAng' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.108991156, 138.170000000]: Key 'vmcKdAng' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.109140206, 138.170000000]: Key 'vmcKiAng' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528186876.109440126, 138.170000000]: Key 'vmcKffAng' was already inserted in ParameterInterface list. Overwriting reference!
[ INFO] [1528186876.110224347, 138.170000000]: [PlannerBase]  Start initialization.
[ INFO] [1528186876.110580148, 138.170000000]: [PlannerBase]  End initialization.
[ INFO] [1528186876.115974726, 138.170000000]: [Signal logger] All done, captain! Stored logging data to file silo_20180605_10-21-16_01344
[ INFO] [1528186876.125140305, 138.175000000]: Started /anymal_ctrl_free_gait/execute_steps action server.
[ INFO] [1528186876.205114367, 138.231000000]: [Rocoma][free_gait_impedance_ros] Add worker free_gait_impedance_ros_publisher!
[ INFO] [1528186876.205317240, 138.231000000]: [Rocoma][free_gait_impedance_ros] Start worker free_gait_impedance_ros_publisher!
[ INFO] [1528186876.205708579, 138.231000000]: Worker [free_gait_impedance_ros_publisher] started
[ INFO] [1528186876.206038548, 138.232000000]: [Rocoma][free_gait_impedance_ros] Successfully swapped!
[ WARN] [1528186877.018002839, 138.562000000]: Computation of publish worker in simulation is not real-time! Computation time: 3.717601 ms
[ INFO] [1528186877.756441155, 138.853000000]: [Signal logger] Use looping Time Buffer of size:1000
[ INFO] [1528186877.756663488, 138.854000000]: [Rocoma] Switched to controller free_gait_impedance_ros
[ INFO] [1528186877.757113618, 138.854000000]: Worker [switch_from_freeze_to_free_gait_impedance_ros] terminated.
[ INFO] [1528186877.765441255, 138.856000000]: waitForService: Service [/free_gait_action_loader/move_base/set_pose] has not been advertised, waiting...
[INFO] [1528186877.766966, 138.856000]: Loading Free Gait action from Python script "/home/remod/projects/anymal/catkin_ws/src/anymal_ctrl_free_gait/anymal_basic_actions/actions/move_base.py".
[ INFO] [1528186877.848785814, 138.885000000]: waitForService: Service [/free_gait_action_loader/move_base/set_pose] is now available.
[INFO] [1528186877.851455, 138.887000]: [FreeGaitActionLoader/MoveBase]: Received move pose request.
[ INFO] [1528186877.852698786, 138.888000000]: Received goal for StepAction.
[INFO] [1528186877.853559, 138.888000]: Action switched to state: Active.
[INFO] [1528186877.856764, 138.888000]: Action switched to state: Pending.
[ INFO] [1528186879.905866280, 139.597000000]: StepAction succeeded.
[INFO] [1528186879.909160, 139.598000]: Action switched to state: Idle.
[ INFO] [1528186879.965243950, 139.616000000]: [mission_states::ros_interface::ActionClient::cancelGoal] Cancelling '/free_gait_action_loader/execute_action'.
[ INFO] [1528186879.965333229, 139.616000000]: [mission_states::ros_interface::ActionClient::cancelGoal] Waiting for action result of '/free_gait_action_loader/execute_action' for inf s ...
[INFO] [1528186879.966626, 139.615000]: [FreeGaitActionLoader/MoveBase]: Stopping.
[INFO] [1528186879.972512, 139.616000]: Action switched to state: Done.
[ERROR] [1528186879.980300, 139.619000]: An exception occurred while loading the action.
Traceback (most recent call last):
  File "/home/remod/projects/anymal/catkin_ws/src/free_gait/free_gait_action_loader/bin/free_gait_action_loader/action_loader.py", line 140, in send_action
    if self.action.state == ActionState.ERROR:
AttributeError: 'NoneType' object has no attribute 'state'
[ERROR] [1528186879.980959, 139.619000]: None
[ERROR] [1528186879.983507, 139.624000]: Exception in your execute callback: 'NoneType' object has no attribute 'wait_for_state'
Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/actionlib/simple_action_server.py", line 289, in executeLoop
    self.execute_callback(goal)
  File "/home/remod/projects/anymal/catkin_ws/src/free_gait/free_gait_action_loader/bin/free_gait_action_loader/action_loader.py", line 74, in _execute_action_callback
    self.action.wait_for_state([ActionState.ERROR, ActionState.DONE])
AttributeError: 'NoneType' object has no attribute 'wait_for_state'

```

If it works, it looks like this:

```
[ INFO] [1528187036.620324505, 202.359000000]: [mission_states::RosInterface::publishProtocolEntry] PoseAdaptation: Started. ()
[ INFO] [1528187036.625064745, 202.362000000]: Worker [switch_from_freeze_to_free_gait_impedance_ros] started
[ WARN] [1528187036.625980618, 202.362000000]: [tinyxml_tools]: Could not find attribute 'maximalNormalForce' in element: <Constraints minimalNormalForce="15.0" />. Setting to default value '1.79769e+308'.
[ WARN] [1528187036.626468561, 202.363000000]: [tinyxml_tools]: Could not find attribute 'maximalNormalForce' in element: <Constraints minimalNormalForce="15.0" />. Setting to default value '1.79769e+308'.
[ WARN] [1528187036.628895572, 202.364000000]: NEW initialize
[ WARN] [1528187036.709248508, 202.368000000]: Key '/loco/ContactInvariantDamper/headingDampingGain' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.709371847, 202.368000000]: Key '/loco/ContactInvariantDamper/lateralDampingGain' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.709787202, 202.368000000]: Key 'vmcKpLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.710095368, 202.368000000]: Key 'vmcKdLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.710181220, 202.368000000]: Key 'vmcKiLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.710548234, 202.368000000]: Key 'vmcKffLin' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.710618133, 202.368000000]: Key 'vmcKpAng' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.710678080, 202.368000000]: Key 'vmcKdAng' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.711241493, 202.368000000]: Key 'vmcKiAng' was already inserted in ParameterInterface list. Overwriting reference!
[ WARN] [1528187036.711918601, 202.368000000]: Key 'vmcKffAng' was already inserted in ParameterInterface list. Overwriting reference!
[ INFO] [1528187036.713465778, 202.371000000]: [PlannerBase]  Start initialization.
[ INFO] [1528187036.713678448, 202.371000000]: [PlannerBase]  End initialization.
[ INFO] [1528187036.715891986, 202.373000000]: [Signal logger] All done, captain! Stored logging data to file silo_20180605_10-23-56_01346
[ INFO] [1528187036.732479888, 202.375000000]: Started /anymal_ctrl_free_gait/execute_steps action server.
[ INFO] [1528187036.802227794, 202.402000000]: [Rocoma][free_gait_impedance_ros] Add worker free_gait_impedance_ros_publisher!
[ INFO] [1528187036.802384591, 202.402000000]: [Rocoma][free_gait_impedance_ros] Start worker free_gait_impedance_ros_publisher!
[ INFO] [1528187036.802565205, 202.402000000]: Worker [free_gait_impedance_ros_publisher] started
[ INFO] [1528187036.802685762, 202.402000000]: [Rocoma][free_gait_impedance_ros] Successfully swapped!
[ WARN] [1528187037.170528674, 202.550000000]: Computation of publish of High-level Controller is not real-time! Computation time: 3.150951 ms. (Warning is throttled: 3s)
[ WARN] [1528187037.538102912, 202.704000000]: Rate 'free_gait_impedance_ros_publisher': Processing took too long (0.034868 s > 0.0333333 s). Number of warnings: 1.
[ INFO] [1528187038.466488417, 203.075000000]: [Signal logger] Use looping Time Buffer of size:1000
[ INFO] [1528187038.467029880, 203.075000000]: [Rocoma] Switched to controller free_gait_impedance_ros
[ INFO] [1528187038.467360371, 203.075000000]: Worker [switch_from_freeze_to_free_gait_impedance_ros] terminated.
[INFO] [1528187038.468555, 203.075000]: Loading Free Gait action from Python script "/home/remod/projects/anymal/catkin_ws/src/anymal_ctrl_free_gait/anymal_basic_actions/actions/move_base.py".
[ INFO] [1528187038.472807390, 203.075000000]: waitForService: Service [/free_gait_action_loader/move_base/set_pose] has not been advertised, waiting...
[ INFO] [1528187038.526623096, 203.095000000]: waitForService: Service [/free_gait_action_loader/move_base/set_pose] is now available.
[INFO] [1528187038.528847, 203.095000]: [FreeGaitActionLoader/MoveBase]: Received move pose request.
[INFO] [1528187038.529970, 203.096000]: Action switched to state: Pending.
[ INFO] [1528187038.531003546, 203.096000000]: Received goal for StepAction.
[INFO] [1528187038.532074, 203.096000]: Action switched to state: Active.
[INFO] [1528187038.579393, 203.113000]: Action was successfully started.
[ WARN] [1528187038.769217232, 203.203000000]: Computation of publish worker in simulation is not real-time! Computation time: 5.126084 ms
[ WARN] [1528187038.878232919, 203.260000000]: Computation of publish worker in simulation is not real-time! Computation time: 3.469264 ms
[ INFO] [1528187040.270880015, 203.812000000]: StepAction succeeded.
[INFO] [1528187040.271629, 203.812000]: Action switched to state: Idle.
[ INFO] [1528187040.338140300, 203.831000000]: [mission_states::ros_interface::ActionClient::cancelGoal] Cancelling '/free_gait_action_loader/execute_action'.
[ INFO] [1528187040.338242684, 203.831000000]: [mission_states::ros_interface::ActionClient::cancelGoal] Waiting for action result of '/free_gait_action_loader/execute_action' for inf s ...
[INFO] [1528187040.338773, 203.831000]: [FreeGaitActionLoader/MoveBase]: Stopping.
[INFO] [1528187040.342640, 203.832000]: Action switched to state: Done.
[ INFO] [1528187040.386046227, 203.850000000]: [mission_states::ros_interface::ActionClient::cancelGoal] The result of '/free_gait_action_loader/execute_action' has been received within 0.019 s.

```

**Comment by FGiraldez** [comment]
@remod should have been fixed by [PR#67](https://github.com/leggedrobotics/free_gait/pull/67). Let me know if you saw it again.

**Comment by remod** [comment]
Let's close it, it did not appear anymore.
